system/Database/Forge.php

```diff
@@ -228,23 +252,48 @@ public function createDatabase(string $db_name): bool
 
 		if (! empty($this->db->dataCache['db_names']))
 		{
-			$this->db->dataCache['db_names'][] = $db_name;
+			$this->db->dataCache['db_names'][] = $dbName;
 		}
 
 		return true;
 	}
 
 	//--------------------------------------------------------------------
 
+	/**
+	 * Determine if a database exists
+	 *
+	 * @param string $dbName
+	 *
+	 * @return boolean
+	 * @throws \CodeIgniter\Database\Exceptions\DatabaseException
+	 */
+	private function databaseExists(string $dbName): bool
+	{
+		if ($this->checkDatabaseExistStr === null)
+		{
+			if ($this->db->DBDebug)
+			{
+				throw new DatabaseException('This feature is not available for the database you are using.');
+			}
+
+			return false;
+		}
+		
+		return $this->db->query(sprintf($this->checkDatabaseExistStr, $this->db->escape($dbName)))->getRow() !== null;
```

Comment:
instead of sprintf here, please use a query with [query bindings](https://codeigniter4.github.io/CodeIgniter4/database/queries.html#query-bindings)

Reply:
ok, rewrote it